### ISSUE TITLE
More Sphinx Work

### ISSUE THREAD

**Issue by jslay88** [ikalchev/HAP-python#49]
Putting this issue here to track progress on the Sphinx Documentation.

Todo:

- We will need to go through more of the project and switch over to sphinx docstrings. I only hit Accessory, Bridge, and AccessoryDriver.
- Ubuntu LTS Installation Pre-Requisites.

**Comment by ikalchev** [comment]
You mean the git pushes?

**Comment by jslay88** [comment]
No, this basically, is for the work that needs to be done still to finish out the documentation so that others can see our needs for documentation, and can also contribute.

The tag will be good for organizational purposes for people looking specifically for documentation work.

**Comment by ikalchev** [comment]
Like so?

**Comment by jslay88** [comment]
Yeah, also, since you are more familiar with the scope of the project, can you please let me, and others here, know what else besides the 3 classes that need to be documented with priority?

**Comment by jslay88** [comment]
Also, it should be ready to be able to hook into readthedocs.io

https://docs.readthedocs.io/en/latest/getting_started.html#sign-up-and-connect-an-external-account

**Comment by ikalchev** [comment]
1. [characteristic](pyhap/characteristic.py)
2. [service](pyhap/service.py)
3. [util](pyhap/util.py) - good to have but not at all that important right now

Yes, I was thinking of readthedocs and will probably do it in the following days.

**Comment by jslay88** [comment]
Keep in mind that the sphinx docstrings are different.

`:param x:`

instead of 

`@param x:`

**Comment by jslay88** [comment]
Doesn't look like the working directory is getting set on rtd. API is not building out the autoclass. Investigating.

**Comment by ikalchev** [comment]
I barely had the time to do the Travis and rtd integrations, hopefully I full find more time soon. Thanks for keeping an eye on that.

**Comment by jslay88** [comment]
No problem. Got it figured it. Will post more info in a few minutes once I have everything nailed down.

**Comment by jslay88** [comment]
Ok, there may be a few settings to change on RTD.

Under the project panel, Admin -> Advanced Settings

- We need to have `Install your project inside a virtualenv using setup.py install` checked.
- `Python interpreter` needs to be set to `CPython3.x`

I will submit a PR for the code changes needed.

**Comment by jslay88** [comment]
Also, these have been documented from the last doc merge.

- Characteristic
- CharLoader
- Service
- ServiceLoader
- Util
